netket/graph/lattice.py

```diff
@@ -176,7 +230,10 @@ def __init__(
                 for atom_coord in atoms_coord
             ]
         )
-        if atoms_coord_fractional.min() < 0 or atoms_coord_fractional.max() >= 1:
```

Comment:
this is about the constructor of Lattice.

If we are moving away from cutoff_tol and towards digits, then we should change this in the constructor too and use the one passed in the constructor throughout the class methods.

However, are we sure we want to use digits instead of tolerance?
We can always extract the number of digits by using a `log10` of the tolerance, and it's more familiar for people to use then the number of digits.

I'd prefer to stick with `cutoff_tol` and compute the number of digits unless there is a compelling reason not to do so.

Reply:
This is fine, we can still with cutoff tol